qemu/tests/block_iscsi_fault_disk.py

```diff
@@ -0,0 +1,157 @@
+        3) Login guest and do io on the disk.
+        4) Kill the qemu process and wait it truly be killed.
+        5. Check the kill time it should less than expected timeout
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def _prepare_fault_disk():
+        cmd = params['cmd_get_scsi_debug']
+        process.run(cmd, shell=True)
+        cmd = "cat " + params['dev_scsi_debug']
+        params['scsi_debug_disk'] = process.getoutput(cmd, shell=True)
+        if not params['scsi_debug_disk']:
+            test.fail("Can not find scsi_debug disk %s" % cmd)
+
+        cmd_dmsetup = """dmsetup create {0} << EOF
+0 160000 linear {1} 0
+160000 5 delay {1} 160000 0 {1} 160000 300000
+160005 80000 linear {1} 160005
+EOF""".format(params['dev_mapper'], params['scsi_debug_disk'])
+        process.run(cmd_dmsetup, shell=True)
+
+        cmd = "dmsetup info " + params['dev_mapper']
+        process.run(cmd, shell=True)
+        params['mapper_disk'] = "/dev/mapper/" + params['dev_mapper']
+        params['emulated_image'] = params['mapper_disk']
+
+    def _cleanup():
+        if vm and vm.is_alive():
+            vm.destroy()
+        if params['mapper_disk']:
+            cmd_cleanup = params['cmd_cleanup']
+            process.run(cmd_cleanup, 600, shell=True)
+
+    def _online_disk_windows(index):
+        disk = "disk_" + ''.join(
+            random.sample(string.ascii_letters + string.digits, 4))
+        online_cmd = "echo select disk %s > " + disk
+        online_cmd += " && echo online disk noerr >> " + disk
+        online_cmd += " && echo clean >> " + disk
+        online_cmd += " && echo attributes disk clear readonly >> " + disk
+        online_cmd += " && echo detail disk >> " + disk
+        online_cmd += " && diskpart /s " + disk
+        online_cmd += " && del /f " + disk
+        return session.cmd(online_cmd % index, timeout=timeout)
+
+    def _get_window_disk_index_by_uid(wwn):
+        cmd = "powershell -command \"get-disk|?"
+        cmd += " {$_.UniqueId -eq '%s'}|select number|FL\"" % wwn
+        status, output = session.cmd_status_output(cmd)
+        if status != 0:
+            test.fail("execute command fail: %s" % output)
+        logging.debug(output)
+        output = "".join([s for s in output.splitlines(True) if s.strip()])
+
+        info = output.split(":")
+        if len(info) > 1:
+            return info[1].strip()
+
+        cmd = "powershell -command \"get-disk| FL\""
+        output = session.cmd_output(cmd)
+        logging.debug(output)
+        test.fail("Not find expected disk:" + wwn)
+
+    def _get_disk_wwn(devname):
+        cmd = "lsblk -ndo WWN " + devname
+        output = process.system_output(cmd, shell=True).decode()
+        wwn = output.replace("0x", "")
+        return wwn
+
+    vm = None
+    iscsi = None
+    params['scsi_debug_disk'] = None
```

Comment:
scsi_debug_disk = params.get("scsi_debug_disk", None)

Reply:
1.This value should not set in cfg
2.It act as "global val" and be shared in inner function